lib/web_catalog/object_graph_importer.es6.js

```diff
@@ -134,34 +134,65 @@ foam.CLASS({
           if (stat.isFile()) {
             if (!ogFileRegExp.test(objectGraphFiles[i])) continue;
 
-            // Drop slice(0, -5): ".json".
-            let releaseInfo = objectGraphFiles[i].slice(0, -5).split('_');
-            let bName = releaseInfo[1];
-            let bVersion = releaseInfo[2];
-            let osName = releaseInfo[3];
-            let osVersion = releaseInfo[4];
+            let releaseInfo = this.getReleaseInfo(objectGraphFiles[i]);
 
             this.info(`Extracting APIs from
-                ${bName} ${bVersion} ${osName} ${osVersion}`);
+                ${releaseInfo.browser.name} ${releaseInfo.browser.version}
+                ${releaseInfo.platform.name} ${releaseInfo.platform.version}`);
             promises.push(
                 this.apiExtractorService.extractWebCatalog(filePath)
                 .then(catalog => {
                   this.info(`Importing APIs
-                      ${bName} ${bVersion} ${osName} ${osVersion}`);
-                  return this.apiImporter.import(bName, bVersion, osName,
-                                                 osVersion, catalog);
+                      ${releaseInfo.browser.name} ${releaseInfo.browser.version}
+                      ${releaseInfo.platform.name}
+                      ${releaseInfo.platform.version}`);
+                  return this.apiImporter.import(
+                      releaseInfo.browser.name, releaseInfo.browser.version,
+                      releaseInfo.platform.name, releaseInfo.platform.version,
+                      catalog);
                 }).then(() => {
                   this.info(`APIs imported for
-                                 ${bName} ${bVersion} ${osName} ${osVersion}`);
+                      ${releaseInfo.browser.name} ${releaseInfo.browser.version}
+                      ${releaseInfo.platform.name}
+                      ${releaseInfo.platform.version}`);
                 }));
           }
         }
         return Promise.all(promises);
       },
     },
+    function getReleaseInfo(filename) {
+      const json = JSON.parse(fs.readFileSync(
+          `${this.objectGraphPath}/${filename}`));
```

Comment:
This should be path.join'd instead of concatenated. I'd also try to pass in the full path instead of the name to minimize the number of places where you are building the path.

Reply:
The downside of that approach is that we then need to extract the filename for (or inside) `getReleaseInfoFromFilename_()`.

Reply:
Calling path.basename(filepath) seems less prone to breakage than building the path in two places.